### DISCUSSION TITLE
Useful cluster commands

### DISCUSSION THREAD

**Discussion by emolinab** [magpiemodel/magpie#424] [General]
The idea of this discussion is to collect helpful cluster commands. Here are a few I used/learned during our landuse folder space "crisis" :

   1) Delete a folder: `rm -rf <FOLDER>`
   2) Copy one file to another folder: `cp file.txt <NEW FOLDER>`
   3) Copy one folder to another: `cp -R <FOLDER> <NEW FOLDER>`
   4) Move one file to another folder: `mv file.txt <NEW FOLDER>`
   5) Move one folder to another folder: `mv <FOLDER> <NEW FOLDER>`
**6) Space available and used in a folder: `mmlsquota -j <FOLDER> --block-size T projects`**
**7) Space occupied in a directory by file and subfolder: `du -h`**
   8) Space occupied per folder and sorted in the current directory:  `du -h -d 1 | sort -rh` (add >> du.txt to save results in a .txt file)

**Comment by pfuehrlich-pik** [comment]
1. Start an interactive shell session on a compute node: `srun --qos=priority --pty bash`
1. Show running processes of a user: `ps -fu <username>`
1. Search all files recursively from the current working directory for a string: `rg <search string>`, use `-Tjson` to *exclude* json files
1. Search all pik-piam repos for a string: `reposearch <search string>`, use `-Tjson` to *exclude* json files
